### DISCUSSION TITLE
Cannot import FastSpeech2

### DISCUSSION THREAD

**Discussion by St33lMouse** [speechbrain/speechbrain#1866] [Q&A]
import torchaudio
from speechbrain.pretrained import FastSpeech2
from speechbrain.pretrained import HIFIGAN

This snippet is from Speechbrain example code. Line 3 is OK, but line 2 gets this error:
ImportError: cannot import name 'fastspeech2' from 'speechbrain.pretrained'

Any idea why?

**Comment by Adel-Moumen** [comment]
Hello,

Could you please share the link of "This snippet is from Speechbrain example code" please ?

**Reply by St33lMouse** [reply]
it's in the tts-fastspeech-ljspeech folder readme:


Perform Text-to-Speech (TTS) with FastSpeech2 
import torchaudio
from speechbrain.pretrained import FastSpeech2
from speechbrain.pretrained import HIFIGAN
 # Intialize TTS (tacotron2) and Vocoder (HiFIGAN)
fastspeech2 = FastSpeech2.from_hparams(source="speechbrain/tts-fastspeech2-ljspeech", savedir="tmpdir_tts")
hifi_gan = HIFIGAN.from_hparams(source="speechbrain/tts-hifigan-libritts-16kHz", savedir="tmpdir_vocoder")
 # Running the TTS
mel_output, durations, pitch, energy = fastspeech2.encode_text(input_text)
 # Running Vocoder (spectrogram-to-waveform)
waveforms = hifi_gan.decode_batch(mel_output)
 # Save the waverform
torchaudio.save('example_TTS.wav', waveforms.squeeze(1), 16000)

If you want to generate multiple sentences in one-shot, you can do in this way: 
from speechbrain.pretrained import FastSpeech2
fastspeech2 = FastSpeech2.from_hparams(source="speechbrain/tts-fastspeech2-ljspeech", savedir="tmpdir_tts")
items = [
"A quick brown fox jumped over the lazy dog",
"How much wood would a woodchuck chuck?",
"Never odd or even"
]
mel_outputs, durations, pitch, energy = fastspeech2.encode_batch(items)

**Reply by Adel-Moumen** [reply]
How did you installed SpeechBrain? Via pip or locally?

**Comment by BenoitWang** [comment]
Hi @St33lMouse @Adel-Moumen ,

Sorry for this inconvenience, our fastspeech2 is not yet merged and not ready to use. We'll only need to do some final reviews, I think we can release it very soon. We'll leave a message here when it's ready.

However, you can find an interface in #1572.

**Comment by BenoitWang** [comment]
Hi @St33lMouse , FastSpeech2 has just been merged, it should now work fine with the interface https://huggingface.co/speechbrain/tts-fastspeech2-ljspeech
